src/main/java/studio/magemonkey/fusion/api/events/services/ProfessionService.java

```diff
@@ -0,0 +1,123 @@
+package studio.magemonkey.fusion.api.events.services;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import studio.magemonkey.codex.CodexEngine;
+import studio.magemonkey.codex.util.messages.MessageData;
+import studio.magemonkey.codex.util.messages.MessageUtil;
+import studio.magemonkey.fusion.api.FusionAPI;
+import studio.magemonkey.fusion.api.events.*;
+import studio.magemonkey.fusion.data.player.FusionPlayer;
+import studio.magemonkey.fusion.data.professions.Profession;
+import studio.magemonkey.fusion.data.recipes.CraftingTable;
+import studio.magemonkey.fusion.util.ExperienceManager;
+import studio.magemonkey.fusion.util.PlayerUtil;
+
+public class ProfessionService {
+
+    /**
+     * Call the ProfessionJoinEvent.
+     * @param professionName The name of the profession.
+     * @param player The player that wants to join the profession.
+     * @param moneyCost The money cost to join the profession.
+     * @param expCost The experience cost to join the profession.
+     */
+    public void callProfessionJoinEvent(String professionName, Player player, double moneyCost, int expCost) {
```

Comment:
I think at this point I want to rename these event service methods. While this is _calling_ the event, it is also doing the action of joining the profession itself. If we could call it `joinProfession` for example, this makes it clear that the profession is being joined. Along the way, we also call the event and conditionally do the join based on that, but the event is not the main goal here. The main goal is the actual joining while the event is a biproduct for anyone hooking in to the API.

Reply:
Finished 👍